ibis/backends/sql/dialects.py

```diff
@@ -68,6 +69,45 @@ class Generator(Postgres.Generator):
         }
 
 
+def _calculate_precision(interval_value: int) -> int:
+    """Calculate interval precision.
+
+    FlinkSQL interval data types use leading precision and fractional-
+    seconds precision. Because the leading precision defaults to 2, we need to
+    specify a different precision when the value exceeds 2 digits.
+
+    (see
+    https://learn.microsoft.com/en-us/sql/odbc/reference/appendixes/interval-literals)
+    """
+    # log10(interval_value) + 1 is equivalent to len(str(interval_value)), but is significantly
+    # faster and more memory-efficient
+    if interval_value == 0:
```

Comment:
`interval_value == 0` covers the case of `ibis.interval(0)`.
`interval_value < 0` should technically never get hit in this context because a negative interval is not allowed, but this is just to be proactive about error-catching and in case this function gets used elsewhere.

Reply:
Not a blocker.